rust/lance/src/index.rs

```diff
@@ -1374,8 +1374,19 @@ impl DatasetIndexInternalExt for Dataset {
                         Box::new(TextQueryParser::new(index.name.clone(), true))
                             as Box<dyn ScalarQueryParser>
                     }
-                    ScalarIndexType::Inverted => Box::new(FtsQueryParser::new(index.name.clone()))
-                        as Box<dyn ScalarQueryParser>,
+                    ScalarIndexType::Inverted => {
+                        let fts_index =
+                            lance_index::scalar::expression::ScalarIndexLoader::load_index(
+                                self,
+                                &field.name,
+                                &index.name,
+                                &NoOpMetricsCollector,
+                            )
+                            .await?;
```

Comment:
This is a bit unique but I think it'll be fine for now.  It means we will always load the FTS index on every query, even if it isn't used.  Maybe once https://github.com/lancedb/lance/pull/4584 we can find someway to handle is_query_allowed in the plugin (perhaps based on details) so we don't have to load an index unless we know we can use it.